samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -287,46 +376,36 @@ SamzaContainer getContainer() {
 
     @Override
     public void onContainerStart() {
+      LOGGER.warn("Received container start notification for container: {} in stream processor: {}.", container, processorId);
       if (!processorOnStartCalled) {
-        // processorListener is called on start only the first time the container starts.
-        // It is not called after every re-balance of partitions among the processors
+        processorListener.onStart();
         processorOnStartCalled = true;
-        if (processorListener != null) {
-          processorListener.onStart();
-        }
-      } else {
-        LOGGER.warn("Received duplicate container start notification for container: {} in stream processor: {}.", container, processorId);
```

Comment:
is this log line not needed? It seems like it'll be useful to log duplicate notifications?

Reply:
`SamzaContainer` triggers `containerListener.onContainerStart` after starting all its components.

When `SamzaContainer` is stopped/started multiple times by `StreamProcessor`, each `containerStart` callback from different `SamzaContainer` instances  are not duplicate. 

The boolean variable guard is to prevent firing multiple `onProcessorStart` callbacks to LocalApplicationRunner layer and that deleted log line was a bug(Existing SamzaContainer implementation doesn't trigger duplicate `onContainerStart` callback within its single lifecycle).